frontendservice/src/views/SignUp.vue

```diff
@@ -0,0 +1,172 @@
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License. */
+
+<template>
+  <div class="sign-up">
+    <h1>Sign Up</h1>
+    <span>New to Virtual Queues? Create an account now!</span>
+    <div class="input-icons">
+      <i class="fa fa-user icon"></i>
+      <input class="input-field" type="text" v-model="name" placeholder="Name" />
+    </div>
+    <p>{{nameAlert}}</p>
+    <div class="input-icons">
+      <i class="fa fa-envelope icon"></i>
+      <input class="input-field" type="text" v-model="email" placeholder="Email" />
+    </div>
+    <p>{{emailAlert}}</p>
+    <div class="input-icons">
+      <i @click="toggle" :class="{'fa fa-eye icon': isHidden, 'fa fa-eye-slash icon': !isHidden}"></i>
+      <input class="input-field" ref="password" type="password" v-model="password" placeholder="Password" />
+    </div>
+    <p>{{passwordAlert}}</p>
+    <a class="button is-primary" @click="createUser">Sign Up</a>
+    <p>{{firebaseAlert}}</p>
+    <div>
+      If you have already registered, go to <router-link to="/login">login</router-link>.
+    </div>
+    <div>
+      Or go back to <router-link to="/about">about</router-link>.
+    </div>
+  </div>
+</template>
+
+<script>
+import firebase from "firebase";
+export default {
+  name: "signUp",
+  data() {
+    return {
+      email: "",
+      password: "",
+      name: "",
+      isHidden: true,
+      nameAlert: "",
+      emailAlert: "",
+      passwordAlert: "",
+      firebaseAlert: ""
+    };
+  },
+  methods: {
+    toggle() {
+      var passwordRef = this.$refs.password;
+      if (passwordRef.type === "password") {
+        passwordRef.type = "text";
+      } else {
+        passwordRef.type = "password";
+      }
+      this.isHidden = !this.isHidden;
+    },
+    createUser: function() {
+      this.nameAlert = "";
+      this.emailAlert = "";
+      this.passwordAlert = "";
+      this.firebaseAlert = "";
+
+      //to check if the input email follows correct format like *@*.*
+      //eg - virtualqueues1@gmail.com is valid
+      var emailRegex = new RegExp(/\w+@\w+.\w+/g);
+      //to check if the input password contains any digit
+      //eg - xyz123 is valid
+      var passwordRegex1 = new RegExp(/[0-9]/g);
+      //to check if the input password contains any special character (!, @, #, %, &, _ or -)
+      //eg - abcde! is valid
+      var passwordRegex2 = new RegExp(/(!|@|#|%|&|_|-)/g);
+      var emailResult = this.email.match(emailRegex);
+      var passwordResult1 = this.password.match(passwordRegex1);
+      var passwordResult2 = this.password.match(passwordRegex2);
+
+      if (this.name.length <= 3 && this.name.length >= 0) {
+        this.nameAlert = "Name must be of atleast 3 characters.";
+      } else if (this.name.length > 30) {
+        this.nameAlert = "Name can have atmost 30 characters only.";
+      }
```

Comment:
Add different methods getNameInputAlertIfInvalid, getEmailInputAlertIfInvalid, etc and return appropriate error string from these methods after performing the required checks in case of input errors.

Reply:
Added.